scripts/UpdateRepos.ps1

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env powershell
+
+<#
+.SYNOPSIS
+    Updates each repo Universe builds to new dependencies.props
+.PARAMETER Source
+    The NuGet package source to find the lineup on.
+.PARAMETER LineupID
+    The ID of the Lineup to determine which versions to use.
+.PARAMETER LineupVersion
+    The version of the Lineup to be used.
+#>
+[cmdletbinding(SupportsShouldProcess = $true)]
+param(
+    [Parameter(Mandatory=$true)]
+    [string]$Source,
+    [Parameter(Mandatory=$true)]
+    [string]$LineupID,
+    [Parameter(Mandatory=$true)]
+    [string]$LineupVersion
+)
+
+$ErrorActionPreference = 'Stop'
+Set-StrictMode -Version 2
+
+Import-Module "$PSScriptRoot/common.psm1" -Scope Local -Force
+
+$RepoRoot = Resolve-Path "$PSScriptRoot\.."
+$ModuleDirectory = Join-Path $RepoRoot "modules"
+
+try {
+    Push-Location $ModuleDirectory
+
+    # Init all submodules
+    Invoke-Block { & git submodule update --init }
+
+    $submodules = Get-Submodules $ModuleDirectory
+
+    $submodules `
+    | % {
+        try {
+            Push-Location $_.path
+
+            $depsFile = Join-Path (Join-Path $($_.path) "build") "dependencies.props"
+            $gitModules = Join-Path $RepoRoot ".gitmodules"
+            # Move to latest commit on tracked branch
+            $branch = $(git config -f $gitModules --get submodule.modules/$($_.module).branch )
+
+            Invoke-Block { & git checkout --quiet $branch }
+
+            Invoke-Block { & .\run.ps1 upgrade deps --source $Source --id $LineupID --version $LineupVersion --deps-file $depsFile }
+            Invoke-Block { & git add $depsFile }
+            Invoke-Block { & git commit --quiet --author "ASP.NET Push Bot <asplab@microsoft.com>" -m "[auto]Update dependencies.props" }
+            $sshUrl = "git@github.com:aspnet/$($_.module)"
+            Invoke-Block { & git remote set-url --push origin $sshUrl }
+
+            # TODO: How to push commits?
+        }
```

Comment:
I'd recommend doing push after you confirmed that you can make changes to the deps.props file successfully. Something like this:

```ps1
$update_errors = @()
foreach ($sm in $submodules) {
   try {
       # (upgrade-dependencies-props)
   }
   catch {
      $update_errors += $_
   }
}

if ($update_errors.Count -gt 0 ) {
  throw 'failed to update'
}

$push_errors = @()
foreach ($sm in $submodules) {
   try {
       # git-push
   }
   catch {
      $push_errors += $_
   }
}
```